Add support for Symfony 7

johanib opened the pull request:
* Add support for SF 7
* Drop support for SF 5
* Upgrade QA tools and rules
* ~Devtooling: remove abandoned phpcbf~

johanib committed:
Update packages

* Add support for SF 7
* Drop support for SF 5
* Upgrade QA tools and rules

MKodde commented on ci/qa/phpcpd:
```diff
@@ -1,6 +0,0 @@
-cd $(dirname $0)/../../
-
-# https://github.com/sebastianbergmann/phpcpd
-vendor/bin/phpcpd ./src
```
See my suggestion on the other bundles; there is a substitution for Sebastians abandoned package

johanib replied:
The shim is the phar 'binary' of the old phpcpd. This solves the composer requirements issue. 
For now, I've used the shim.

johanib replied:
Good suggestion btw, I did not know :1st_place_medal:

MKodde commented on ci/qa/phpstan.neon:
```diff
@@ -1,9 +1,8 @@
     - ../../vendor/phpstan/phpstan-deprecation-rules/rules.neon
 parameters:
-	level: 2
+	level: 10
```
Brave!

MKodde commented on CHANGELOG.md:
```diff
@@ -1,5 +1,14 @@
 # Changelog
 
+## 4.3.0
```
Should this not be a major? Given you drop support for a SF version?

MKodde commented on composer.json:
```diff
@@ -63,6 +65,9 @@
             "dealerdirect/phpcodesniffer-composer-installer": true
+        },
+        "platform": {
+          "php": "8.2.0"
```
I think you better not bind to a specific platform version in a bundle

johanib replied:
:+1: 

I agree. Should not be used for libraries.

https://getcomposer.org/doc/06-config.md#platform

johanib committed:
Rework

- symfony/console should not lock this library to SF 7
- Add missing licence headers

MKodde reviewed (approved):
Truly a PR in the spirit of the boy scout rule.

🌲
